components/org.wso2.carbon.identity.api.idpmgt/src/main/java/org/wso2/carbon/identity/api/idpmgt/IDPMgtBridgeService.java

```diff
@@ -0,0 +1,736 @@
+
+        IdentityProvider idp;
+        try {
+            idp = identityProviderManager.getIdPByName(idpName, tenantDomain);
+        } catch (IdentityProviderManagementException e) {
+            throw handleServerException(ErrorMessages.ERROR_CODE_INTERNAL_ERROR, null, e);
+        }
+        if (idp == null) {
+            throw handleClientException(ErrorMessages.ERROR_CODE_RESOURCE_NOT_FOUND, null);
+        }
+
+        return idp;
+    }
+
+    /**
+     * Function returns the set of authenticators for a given IDP
+     *
+     * @param idpName name of the idp
+     * @param limit   limit of the query
+     * @param offset  starting index of the query
+     * @return list of authenticators which satisfy the query parameters
+     * @throws IDPMgtBridgeServiceException throws
+     */
+    public List<FederatedAuthenticatorConfig> getAuthenticatorList(String idpName, Integer limit, Integer offset) throws
+            IDPMgtBridgeServiceException {
+
+        IdentityProvider idp = getIDPByName(idpName, getTenantDomain());
+
+        if (ArrayUtils.isNotEmpty(idp.getFederatedAuthenticatorConfigs())) {
+            List<FederatedAuthenticatorConfig> federatedAuthenticatorConfigs = Arrays.asList(idp
+                    .getFederatedAuthenticatorConfigs());
+
+            Pair<Integer, Integer> indexValue = paginationList(federatedAuthenticatorConfigs.size(), limit, offset);
+            federatedAuthenticatorConfigs.sort(Comparator.comparing(FederatedAuthenticatorConfig::getName));
+            return federatedAuthenticatorConfigs.subList(indexValue.getKey(), indexValue.getValue());
+        } else {
+            return new ArrayList<>();
+        }
+    }
+
+    /**
+     * Function returns the authenticator detail for a given IDP
+     *
+     * @param idpName  name of the idp
+     * @param authName name of the authenticator
+     * @return list of authenticators which satisfy the query parameters
+     * @throws IDPMgtBridgeServiceException throws
+     */
+    public List<FederatedAuthenticatorConfig> getAuthenticatorByName(String idpName, String authName)
+            throws
+            IDPMgtBridgeServiceException {
+
+        IdentityProvider idp = getIDPByName(idpName, getTenantDomain());
+
+        if (ArrayUtils.isNotEmpty(idp.getFederatedAuthenticatorConfigs())) {
+            List<FederatedAuthenticatorConfig> federatedAuthenticatorConfigs = Arrays.asList(idp
+                    .getFederatedAuthenticatorConfigs());
+            Optional<FederatedAuthenticatorConfig> optionalFederatedAuthenticatorConfig = federatedAuthenticatorConfigs
+                    .stream().filter
+                            (federatedAuthenticatorConfig ->
+                                    federatedAuthenticatorConfig.getName().equals(authName)).findFirst();
+            if (optionalFederatedAuthenticatorConfig.isPresent()) {
+                ArrayList<FederatedAuthenticatorConfig> federatedAuthenticatorConfigs1 = new ArrayList<>();
```

Comment:
federatedAuthenticatorConfigs1 doesn't look like a meaningful name. Why do we need 1 at the end ?